kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstance.java

```diff
@@ -90,25 +100,51 @@ public void updated(Map<String, Object> properties) {
         }
 
         try {
-            ContainerInstanceOptions newProps = new ContainerInstanceOptions(properties);
-
-            if (newProps.getSignatureTrustAnchor().isPresent()) {
-                ValidationResult containerSignatureValidated = validateContainerImageSignature(newProps);
-                String imageDigest = containerSignatureValidated.imageDigest().orElse("?");
-                logger.info("Container signature validation result for {}@{}({}) - {}", newProps.getContainerImage(),
-                        imageDigest, newProps.getContainerImageTag(),
-                        containerSignatureValidated.isSignatureValid() ? "OK" : "FAIL");
-            } else {
-                logger.info("No trust anchor available. Signature validation skipped.");
+            ContainerInstanceOptions newOptions = new ContainerInstanceOptions(properties);
+
+            if (this.currentOptions != null && this.currentOptions.equals(newOptions)) {
+                return;
             }
 
-            if (newProps.isEnabled()) {
+            this.currentOptions = newOptions;
+
+            if (!this.currentOptions.getEnforcementDigest().isPresent()) {
+
+                logger.info(
+                        "Container configuration doesn't include enforcement digest. Validating with Container Signature Validation service");
+
+                if (this.currentOptions.getSignatureTrustAnchor().isPresent()) {
+
+                    ValidationResult containerSignatureValidated = validateContainerImageSignature(this.currentOptions);
+
+                    logger.info("Container signature validation result for {}@{}({}) - {}",
+                            this.currentOptions.getContainerImage(),
+                            this.currentOptions.getEnforcementDigest().orElse("?"),
```

Comment:
If we want to display the resulting digest we should log `containerSignatureValidated`... not the current options.

```suggestion
                            containerSignatureValidated.imageDigest().orElse("?"),
```